core/block.js

```diff
@@ -1335,3 +1335,37 @@ Blockly.Block.prototype.moveBy = function(dx, dy) {
 Blockly.Block.prototype.makeConnection_ = function(type) {
   return new Blockly.Connection(this, type);
 };
+
+/**
+ * Recursively checks whether all statement and value inputs are filled with
+ * blocks. Also checks all following statement blocks in this stack.
+ * @param {boolean=} opt_shadowBlocksAreFilled An optional argument controlling
+ *     whether shadow blocks are counted as filled. Defaults to true.
+ * @return {boolean} True if all inputs are filled, false otherwise.
+ */
+Blockly.Block.prototype.allInputsFilled = function(opt_shadowBlocksAreFilled) {
+  // Handle the default value for the optional argument.
+  if (opt_shadowBlocksAreFilled === undefined) {
+    opt_shadowBlocksAreFilled = true;
+  }
+
```

Comment:
Instead of checking on connected blocks, just check on the current block:
```
if (this.isShadow() && !opt_shadowBlocksAreFilled) {
    return false;
}
```